src/modules/app/onBeforeChange.js

```diff
@@ -2,53 +2,32 @@
 
 import type { Dispatch, GetState, Bag } from 'redux-first-router/dist/flow-types'
 import { CATEGORIES_ROUTE } from './routes/categories'
-import { clearStoreOnCityChange, clearStoreOnLanguageChange } from '../endpoint/actions/remover'
-import citiesFetcher from '../endpoint/fetchers/cities'
-import languagesFetcher from '../endpoint/fetchers/languages'
-import eventsFetcher from '../endpoint/fetchers/events'
+import citiesEndpoint from '../endpoint/endpoints/cities'
+import languagesEndpoint from '../endpoint/endpoints/languages'
+import eventsEndpoint from '../endpoint/endpoints/events'
 
 import { EVENTS_ROUTE } from './routes/events'
 import { EXTRAS_ROUTE } from './routes/extras'
 import { DISCLAIMER_ROUTE } from './routes/disclaimer'
 import { SEARCH_ROUTE } from './routes/search'
 
 /**
- * This handles the use of the back/ next button of the browser. We have to remove and refetch the categories in the old
- * language, because the stopover of CategoriesRedirect, during which the refetch normally happens, is not executed again
- * when using the back/ next button.
- * @param dispatch
- * @param getState
- * @param bag
+ * This handles the loading of additional data for the location layout
+ * (onBeforeChange is executed before a change of the route)
  */
 const onBeforeChange = async (dispatch: Dispatch, getState: GetState, bag: Bag) => {
   const state = getState()
   const {city, language} = bag.action.payload
   const route = bag.action.type
-  const prevCity = state.location.payload.city
-  const prevLanguage = state.location.payload.language
   const params = {city, language}
 
-  if (prevLanguage && prevLanguage !== language) {
-    clearStoreOnLanguageChange(dispatch, getState)
-  }
-
-  if (prevCity && prevCity !== city) {
-    clearStoreOnCityChange(dispatch, getState)
-  }
-
   // in the following routes we have a location layout, so we need cities, languages and events
   if ([CATEGORIES_ROUTE, EVENTS_ROUTE, EXTRAS_ROUTE, DISCLAIMER_ROUTE, SEARCH_ROUTE].includes(route)) {
-    if (!state.cities) {
-      await citiesFetcher(dispatch, params)
-    }
+    await citiesEndpoint.loadData(dispatch, state.cities, params)
 
-    if (!state.languages) {
-      await languagesFetcher(dispatch, params)
-    }
+    await languagesEndpoint.loadData(dispatch, state.languages, params)
 
-    if (!state.events) {
-      await eventsFetcher(dispatch, params)
-    }
+    await eventsEndpoint.loadData(dispatch, state.events, params)
```

Comment:
We should load them all in parallel, I think this should work:
```js
  await Promise.all([
    citiesEndpoint.loadData(dispatch, state.cities, params),
    languagesEndpoint.loadData(dispatch, state.languages, params),
    eventsEndpoint.loadData(dispatch, state.events, params)
  ])
```

Reply:
good suggestion